internal/schemautil/custom_diff.go

```diff
@@ -173,3 +175,70 @@ func CustomizeDiffCheckStaticIpDisassociation(_ context.Context, d *schema.Resou
 	// TODO: Check that we block deletions that will result in too few static ips for the plan
 	return nil
 }
+
+// CustomizeDiffDisallowMultipleManyToOneKeys checks that we don't have multiple keys that are going to be converted to
+// a single key in the API request, e.g. 'ip_filter' and 'ip_filter_object' in the same diff.
+func CustomizeDiffDisallowMultipleManyToOneKeys(_ context.Context, d *schema.ResourceDiff, _ interface{}) error {
+	ks := d.GetRawConfig().AsValueMap()
+
+	tks := map[string]struct{}{}
```

Comment:
I think this can be a `[]string` string slice since we only use it as a storage for keys